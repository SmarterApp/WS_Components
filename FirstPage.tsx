import * as React from 'react';

interface Props {
    subject: string;
    grade: string;
}

export class FirstPage extends React.Component<Props, {}> {
    render() {
        const today = new Date();
        return (
            <div className="page first-page">
                <div className="first-page-title">
                    Smarter Balanced<br />
                    Assessment Consortium
                </div>
                <div className="first-page-subtitle">
                    {this.props.subject} Practice Test Scoring Guide<br />
                    <strong>{this.props.grade}</strong>
                </div>
                <div className="first-page-date">
                    {today.getMonth()}/{today.getDate()}/{today.getFullYear()}
                </div>
            </div>
        );
    }
}