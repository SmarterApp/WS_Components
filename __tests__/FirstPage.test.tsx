import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { FirstPage } from '../FirstPage';
import { shallow } from 'enzyme';

const date = {
    getMonth: () => "01",
    getDate: ()  => "12",
    getFullYear:  () => "2017"
}

describe("First page", () => {
    it("matches snapshot", () => {
        let wrapper = shallow(<FirstPage subject="math" grade="grade 3" date={date}/>);
        expect(wrapper).toMatchSnapshot();
    })
})