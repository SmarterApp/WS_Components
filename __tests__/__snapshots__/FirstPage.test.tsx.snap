// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceStatement matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="page first-page"
>
    <div
        className="first-page-title"
    >
        Smarter Balanced
        <br />
        Assessment Consortium
    </div>
    <div
        className="first-page-subtitle"
    >
        math
         Practice Test Scoring Guide
        <br />
        <strong>
            grade 3
        </strong>
    </div>
    <div
        className="first-page-date"
    >
        9
        /
        3
        /
        2017
    </div>
</div>,
  "nodes": Array [
    <div
      className="page first-page"
>
      <div
            className="first-page-title"
      >
            Smarter Balanced
            <br />
            Assessment Consortium
      </div>
      <div
            className="first-page-subtitle"
      >
            math
             Practice Test Scoring Guide
            <br />
            <strong>
                  grade 3
            </strong>
      </div>
      <div
            className="first-page-date"
      >
            9
            /
            3
            /
            2017
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <FirstPage
        grade="grade 3"
        subject="math"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": FirstPage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "grade": "grade 3",
          "subject": "math",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="page first-page"
>
          <div
                    className="first-page-title"
          >
                    Smarter Balanced
                    <br />
                    Assessment Consortium
          </div>
          <div
                    className="first-page-subtitle"
          >
                    math
                     Practice Test Scoring Guide
                    <br />
                    <strong>
                              grade 3
                    </strong>
          </div>
          <div
                    className="first-page-date"
          >
                    9
                    /
                    3
                    /
                    2017
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="page first-page"
>
          <div
                    className="first-page-title"
          >
                    Smarter Balanced
                    <br />
                    Assessment Consortium
          </div>
          <div
                    className="first-page-subtitle"
          >
                    math
                     Practice Test Scoring Guide
                    <br />
                    <strong>
                              grade 3
                    </strong>
          </div>
          <div
                    className="first-page-date"
          >
                    9
                    /
                    3
                    /
                    2017
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FirstPage
    grade="grade 3"
    subject="math"
/>,
}
`;
